### PR TITLE
Extract common WaitState functions 🔨

### PR DESCRIPTION
**Pull request by vdemeester**
It reduces some noise and duplication in the tests 👼

/cc @shashwathi @bobcatfish 

Signed-off-by: Vincent Demeester <vdemeest@redhat.com>

### CODE REVIEW

File: test/crd_checks.go [hunk 1]
```diff
@@ -104,3 +105,59 @@ func WaitForServiceExternalIPState(c *clients, namespace, name string, inState f
 		return inState(r)
 	})
 }
+
+func taskRunSucceed(name string) func(tr *v1alpha1.TaskRun) (bool, error) {
```
Comment (bobcatfish) [inline test/crd_checks.go]: hm i guess it's not required but i feel like these should be Public and have docstrings :thinking:

**Comment by bobcatfish** [comment]
This could also be an opportunity to start moving some of the docs on these libs out of `test/README.md` into godocs :innocent: 

e.g. https://github.com/knative/build-pipeline/tree/master/test#poll-pipeline-resources could be a comment inside `crd_checks.go`....

File: test/crd_checks.go (hunk 1 above)
Comment (vdemeester) [reply test/crd_checks.go]: Fair enough :angel: I did the *smallest* change possible, but I'm definitely +:100: on the docstring :angel: Will update :wink:

**Pull request merged**